nncf/onnx/statistics/collectors.py

```diff
@@ -70,7 +77,16 @@ def masked_mean(
         if mask is None:
             return cls.mean(x, axis=axis, keepdims=keepdims)
         masked_x = np.ma.array(x.tensor, mask=mask.tensor)
-        return ONNXNNCFTensor(np.ma.mean(masked_x, axis=axis, keepdims=False).data)
+        if x.tensor.dtype in [np.float32, np.float16]:
+            result = np.ma.mean(masked_x, axis=axis, keepdims=keepdims, dtype=np.float64)
+            if isinstance(result, np.ma.MaskedArray):
+                result = result.data
+            return ONNXNNCFTensor(result.astype(dtype=x.tensor.dtype))
+        else:
```

Comment:
Can we use `dtype` as a parameter and reduce amount of branches here?

Reply:
The problem is that cast result to type of input tensor required only for float16 and float32, for integer cast is no need, because result is float64. Еhere is no way without using conditions.